streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -92,7 +92,8 @@ private JoinWindows(final long beforeMs,
         this.maintainDurationMs = maintainDurationMs;
     }
 
-    @SuppressWarnings("deprecation") // removing segments from Windows will fix this
+    @SuppressWarnings("DeprecatedIsStillUsed")
```

Comment:
Who defined this string?

Reply:
IntelliJ.

Reply:
The naming convention is a bit broken (e.g it starts with a capital). Generally, it's a bit odd to be adding IDE specific values that are not standardized. For example, would we do the same for Eclipse, Atom, Visual Studio.net, etc.?

Reply:
I guess that's fair -- nobody complained to far -- we use multiple IntelliJ related suppression -- there is no guideline -- happy to define one -- if we don't want to use IntelliJ specific one, we should remove all of them.

\cc @guozhangwang @bbejeck @vvcephei @ableegoldman

Reply:
Fair point, I don't have an issue with it annotation itself, but maybe we are better off with not using an IDE specific value.  Just my 2 cents though, I'll go with the majority on this one.

Reply:
sgtm.

Reply:
@guozhangwang What does "sgtm" mean in this context? Keep the annotation or remove it?

Reply:
Ping @mjsax @guozhangwang  this is ready for merging pending resolving this last comment regarding the annotations.

Reply:
It's still unclear to me how to proceed:
1. remove IntelliJ suppressions?
1a. define guidelines that only Java annotation should be use?
2. keep the annotations?
2a. allow for other annotations of other IDEs?

\cc @ijuma @guozhangwang

Reply:
@mjsax my `sgtm` refers to 1 + 1.a above (not a strong preference though) :) I've also synced with @ijuma just now, and he's fine with 2 if we makes sure 2.a is done and documented clearly.

So I'd suggest making a call yourself given neither of us having a strong preference and proceed.